### PR TITLE
Add option to no-op if target branch is up-to-date.

### LINKED ISSUE #82: Add an option to succeed if there are no changes.
When using git-pull-request in automation, it would be handy to have a mode that is a no-op if there are no changes from the target branch.

### PR DESCRIPTION
**Pull request by tomprince**
Fixes #82.

### CODE REVIEW

**Comment by tomprince** [comment]
>     1. when is that useful?

I've got some code in automation to port changes between two repositories, and then create a pull-request with the result. There sometimes isn't any changes that would result, in which case the code creates a branch that matches the upstream branch.

`git-pull-request` does all the work necessary to make a pull-request to bring the target branch up-to-date with the current branch, except in the case where it is already up-to-date (in which case it fails). I'd like to have an option where it will handle that case, but still error out if there are other problems creating the pull request.

>     2. the name of the option sounds confusing
I don't have strong feelings on the name, and agree that it isn't the clearest. I didn't have any better ideas, though.

**Comment by jd** [comment]
It sounds like you can already do this by catching the exit code `50` in your calling program/script, no?

File: git_pull_request/__init__.py [hunk 1]
```diff
@@ -582,10 +585,16 @@ def fork_and_push_pull_request(g, hosttype, repo_to_fork, rebase,
                                             title=title,
                                             body=message)
         except github.GithubException as e:
-            LOG.critical(
-                _format_github_exception("create pull request", e)
-            )
-            return 50
+            if allow_noop and any((
```
Comment (jd) [inline git_pull_request/__init__.py]: I think it's fine to fail with that. I'd just not add a new option for that.

**Comment by jd** [comment]
No update in a while, closing.

**Pull request closed without merging**